src/runtime_src/driver/hw_em/generic_pcie_hal2/debug.cxx

```diff
@@ -142,7 +142,11 @@ namespace xclhwemhal2 {
           if (count >= size) break;
           if (map->m_debug_ip_data[i].m_type == type) {
             if (baseAddress) baseAddress[count] = map->m_debug_ip_data[i].m_base_address;
-            if (portNames)   portNames[count]   = (char*)map->m_debug_ip_data[i].m_name;
+            if (portNames) {
+               char nameBuf[129] = {0};
+               std::memcpy(nameBuf, (char*)(map->m_debug_ip_data[i].m_name), 128);
+               portNames[count] = nameBuf;
+            }
```

Comment:
I suggest
portNames[count].assign(reinterpret_cast<const char*>(map->m_debug_ip_data[i],m_name,128);

Reply:
Hi Soren, I have updated the pull request with your suggested change.
Thanks for your feedback!